### DISCUSSION TITLE
slash commands work only with (await bot.register_commands(guild_id=guild.id))

### DISCUSSION THREAD

**Discussion by xa1era** [Pycord-Development/pycord#1178] [Q&A] [answered]
I want to make slash commands, but i have many problems:
slash_commands doesn't work with scopes : bot,application.commands on developer.discord
bot.all_comands returning {None: <discord.commands.SlashCommand name=joined>}
i tried to set guild_ids in @bot.slash_command()
also i tried to kick from guild and re-join to guild
but slash commands works only with (await bot.register_commands(guild_id=guild.id))
i used code from [https://github.com/Pycord-Development/pycord/blob/master/examples/app_commands/slash_basic.py](url)
my code
```
import discord

intents = discord.Intents(
    guilds=True,
    members=True,
    messages=True,
)

bot = discord.Bot(intents=intents)

@bot.slash_command(name="hi")
async def global_command(ctx, num: int):
    await ctx.respond(f"This is a global command, {num}!")


@bot.slash_command()
async def joined(ctx, member: discord.Member = None):
    user = member or ctx.author
    await ctx.respond(f"{user.name} joined at {discord.utils.format_dt(user.joined_at)}")

@bot.event  
async def on_connect():
    print('connected')
    print(bot.all_commands)

bot.run(token)
```

**Comment by NeloBlivion** [comment, marked as answer]
You overwrote on_connect, which normally handles command syncing. In `on_connect`, you should add `await bot.sync_commands()`
Also bear in mind global commands can take up to an hour to register.

**Reply by xa1era** [reply]
await bot.sync_commands() adds commands on guilds but bot.all_comands returning {None: <discord.commands.SlashCommand name=joined>, '954044880315908117': <discord.commands.SlashCommand name=joined>, '954052709923815534': <discord.commands.SlashCommand name=hi>}
and only one command works on guild (joined)

**Reply by NeloBlivion** [reply]
if you only just added the sync_commands line, then again you pretty much have to wait out the hour since you've defined them as global commands. If you're just testing out for now, i recommend setting `guild_ids` for your slash commands to bypass that restriction and sync immediately, or to keep it simpler pass `debug_guilds=[guild_id_here]` into `discord.Bot` to apply that to all your commands.

**Reply by xa1era** [reply]
With setting guild_ids commands work, but in guides commands updated without await bot.sync_commands().
Thanks for your help

**Reply by Dorukyum** [reply]
> but in guides commands updated without await bot.sync_commands()

Which "guides", may I ask?
